File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        unordered_map<int, int> mp;
        for (int &num : nums) {
            if (mp[num] == 0) {
                mp[num]++;
            } else {
                return num;
            }
        }
        return -1;
    }
};

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        n = len(nums)
        for i in range(n):
            index = abs(nums[i]) - 1
            if nums[index] < 0:
                return abs(nums[i])
            nums[index] = nums[index] * (-1)
        return -1

# class Solution:
#     def findDuplicate(self, nums: List[int]) -> int:
#         mp = defaultdict(int)
#         for num in nums:
#             mp[num] += 1
#             if mp[num] > 1:
#                 return num
#         return -1